### PR TITLE
Lots more linting, switch to a whitelist approach, fix a small bug in title

### PR DESCRIPTION
**Pull request by euank**
There's probably at least a 50% chance this breaks *something* because this was half human half automated refactoring.

### CODE REVIEW

File: bot.test.js [hunk 1]
```diff
@@ -24,7 +26,7 @@ describe('bot.getReply', () => {
       [['x\ny'], 'x\ny'], // line splitting happens a layer below, we gucci
     ];
 
-    for (let i = 0; i < tests.length; i++) {
+    for (let i = 0; i < tests.length; i += 1) {
```
Comment (LinuxMercedes) [inline bot.test.js]: What

Comment (euank) [reply bot.test.js]: eslint

File: modules/quo/index.js [hunk 2]
```diff
@@ -22,29 +22,39 @@ module.exports.run = function (r, parts, reply, command, from, to) {
...
+      return;
+    }
+    if (res.match(/(pls|#)noquo/)) {
+      reply("don't be a deck, betch");
+      return;
+    }
+
+
+    const replyPromises = quoModules.map(el => new Promise(((resolve) => {
+      el.post(res, to, (err2, info) => {
```
Comment (LinuxMercedes) [inline modules/quo/index.js]: `err2`...I guess it keeps it from shadowing the err in the surrounding scope...

File: bot.test.js (hunk 1 above)
Comment (euank) [reply bot.test.js]: Specifically https://eslint.org/docs/rules/no-plusplus if you're curious their justification. (automatic semicolon insertion ftw)

File: modules/roll/index.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 const Roll = require('roll');
 
-const regex = /^(\d*)d(\d+|\%)(([\+\-\/\*b])(\d+))?$/;
+const regex = /^(\d*)d(\d+|%)(([+\-/*b])(\d+))?$/;
```
Comment (LinuxMercedes) [inline modules/roll/index.js]: We should verify that this works the way we want it to.

**Comment by LinuxMercedes** [comment]
When this merges, we should update #144 to reflect changes.

File: modules/roll/index.js (hunk 3 above)
Comment (euank) [reply modules/roll/index.js]: I tossed in a commit with unit tests for roll and the tests still pass so I'm pretty sure it's okay :)

**Pull request merged**